src/proto/streams/state.rs

```diff
@@ -57,22 +57,22 @@ enum Inner {
     // TODO: these states shouldn't count against concurrency limits:
     //ReservedLocal,
     ReservedRemote,
-    Open { local: Peer, remote: Peer },
-    HalfClosedLocal(Peer), // TODO: explicitly name this value
-    HalfClosedRemote(Peer),
-    // When reset, a reason is provided
-    Closed(Option<Cause>),
+    Open { local: OpenPeer, remote: OpenPeer },
+    HalfClosedLocal(OpenPeer), // TODO: explicitly name this value
+    HalfClosedRemote(OpenPeer),
+    Closed(Cause),
 }
 
 #[derive(Debug, Copy, Clone)]
-enum Peer {
+enum OpenPeer {
     AwaitingHeaders,
     Streaming,
 }
 
 #[derive(Debug, Copy, Clone)]
 enum Cause {
-    Proto(Reason),
+    EndStream,
+    Proto(Peer, Reason),
```

Comment:
Instead of adding a `Peer` variant, could we add a `Cause::LocallyReset` variant. I believe that is the case that we really care about?